packages/synapse-react-client/src/components/widgets/facet-nav/useFacetPlots.ts

```diff
@@ -0,0 +1,96 @@
+import {
+  FacetColumnRequest,
+  FacetColumnResult,
+  FacetColumnResultValueCount,
+  FacetColumnResultValues,
+  QueryResultBundle,
+} from '@sage-bionetworks/synapse-types'
+import { isSingleNotSetValue } from '../../../utils/functions/queryUtils'
+import { tableQueryDataAtom } from '../../QueryWrapper/QueryWrapper'
+import { useAtomValue } from 'jotai'
+import { useMemo, useEffect, useCallback } from 'react'
+import { UniqueFacetIdentifier } from '../../../utils/types/UniqueFacetIdentifier'
+import { FacetNavPanelProps } from './FacetNavPanel'
+import FacetPlotsCard from '../../FeaturedDataTabs/FacetPlotsCard'
+import { useQueryContext } from '../../QueryContext'
+import { applyChangesToValuesColumn } from '../query-filter/FacetFilterControls'
+
+export default function useFacetPlots(
+  facetsToPlot: string[],
+): Pick<
+  FacetNavPanelProps,
+  'applyChangesToFacetFilter' | 'applyChangesToGraphSlice' | 'facetToPlot'
+>[] {
+  const data = useAtomValue(tableQueryDataAtom)
+  const { getCurrentQueryRequest, executeQueryRequest, error } =
+    useQueryContext()
+
+  const lastQueryRequest = useMemo(
+    () => getCurrentQueryRequest(),
+    [getCurrentQueryRequest],
+  )
```

Comment:
This should always be memoized, but it probably isn't in many places across our code base. Let's file a ticket to look into these.

Reply:
https://sagebionetworks.jira.com/browse/PORTALS-2921